src/main/java/fr/insee/genesis/controller/rest/ResponseController.java

```diff
@@ -128,99 +98,65 @@ public ResponseEntity<Object> saveResponsesFromXmlFile(     @RequestParam("pathL
-                    //DOM method
-                    LunaticXmlDataParser parser = new LunaticXmlDataParser();
-                    campaign = parser.parseDataFile(filepath);
-
-                    List<SurveyUnitUpdateDto> suDtos = new ArrayList<>();
-                    for (LunaticXmlSurveyUnit su : campaign.getSurveyUnits()) {
-                        suDtos.addAll(LunaticXmlAdapter.convert(su, variablesMap, campaign.getIdCampaign(),currentMode));
-                    }
-                    surveyUnitQualityService.verifySurveyUnits(suDtos,variablesMap);
-
-                    log.info("Saving {} survey units updates", suDtos.size());
-                    surveyUnitService.saveSurveyUnits(suDtos);
-                    log.info("Survey units updates saved");
-                }else{
-                    //Sequential method
-                    log.warn("File size > " + Constants.MAX_FILE_SIZE_UNTIL_SEQUENTIAL + "MB! Parsing XML file using sequential method...");
-                    try(final InputStream stream = new FileInputStream(filepath.toFile())){
-                        LunaticXmlDataSequentialParser parser = new LunaticXmlDataSequentialParser(filepath,stream);
-                        int suCount = 0;
-
-                        campaign = parser.getCampaign();
-                        LunaticXmlSurveyUnit su = parser.readNextSurveyUnit();
-
-                        while(su != null){
-                            List<SurveyUnitUpdateDto> suDtos = new ArrayList<>(LunaticXmlAdapter.convert(su, variablesMap, campaign.getIdCampaign(),currentMode));
-
-                            surveyUnitQualityService.verifySurveyUnits(suDtos,variablesMap);
-                            surveyUnitService.saveSurveyUnits(suDtos);
-                            suCount++;
-
-                            su = parser.readNextSurveyUnit();
-                        }
-
-                        log.info("Saving {} survey units updates", suCount);
-                        log.info("Survey units updates saved");
-                    }
+                List<Mode> modesList = controllerUtils.getModesList(campaignName, null); //modeSpecified null = all modes
+                for (Mode currentMode : modesList) {
+                    treatCampaignWithMode(campaignName, currentMode, errors);
                 }
-                log.info("File {} saved", fileName);
-                fileUtils.moveDataFile(campaignName, currentMode.getFolder(), fileName);
+            } catch (Exception e) {
+                log.warn("Error for campaign " + campaignName + ": " + e.getMessage());
+                errors.add(new GenesisError(e.getMessage()));
             }
         }
-        return new ResponseEntity<>("Data saved", HttpStatus.OK);
+        if (errors.isEmpty()) {
+            return new ResponseEntity<>("Data saved", HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>("Data saved with " + errors.size() + " errors", HttpStatus.OK);
```

Comment:
ResponseEntity.ok, but if there are errors maybe ok is a bad status... 206 | Partial Content ?